js/views/file_input_view.js

```diff
@@ -132,8 +136,8 @@
                 case 'audio': this.addThumb('images/audio.svg'); break;
                 case 'video': this.addThumb('images/video.svg'); break;
                 case 'image':
-                    this.oUrl = URL.createObjectURL(file);
-                    this.addThumb(this.oUrl);
+                    window.autoOrientImage(file)
+                        .then(dataURL => this.addThumb(dataURL));
```

Comment:
TODO: Test with non-JPEGs.
Nice-to-have: Skip for non-JPEG files since they don’t have EXIF information.